src/generate/register.rs

```diff
@@ -720,13 +727,24 @@ pub fn fields(
                 });
             }
 
-            proxy_items.push(quote! {
-                ///Writes raw bits to the field
-                #[inline(always)]
-                pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
-                    self.w.bits &= !(#mask << #offset);
-                    self.w.bits |= ((value as #rty) & #mask) << #offset;
-                    self.w
+            proxy_items.push(if offset != 0 {
+                let offset = &f.offset;
+                quote! {
+                    ///Writes raw bits to the field
+                    #[inline(always)]
+                    pub #unsafety fn #bits(self, value: #fty) -> &'a mut W {
+                        self.w.bits = self.w.bits & !(#mask << #offset) | ((value as #rty) & #mask) << #offset;
```

Comment:
What do you think about adding some extra parenthesis to make it clearer what the operator precedence is? This is basically `x = x & !y | z << w` which is something of an exercise to parse in your head...

Reply:
Uhm, I could do but would it actually make a difference? And where would you like them? ;)

Reply:
The current logic is correct so I suggest adding them to maintain correctness :p 

```rust
self.w.bits = (self.w.bits & !(#mask << #offset)) | (((value as #rty) & #mask) << #offset);
```

Reply:
Done.